tests/Doctrine/Tests/ORM/Functional/Ticket/GH7062Test.php

```diff
@@ -0,0 +1,255 @@
+<?php
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH7062Test extends OrmFunctionalTestCase
+{
+    private const SEASON_ID = 'season_18';
+    private const TEAM_ID   = 'team_A';
+
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema(
+            [
+                GH7062Team::class,
+                GH7062Season::class,
+                GH7062Ranking::class,
+                GH7062RankingPosition::class
+            ]
+        );
+    }
+
+    /**
+     * @group 7062
+     */
+    public function testEntityWithAssociationKeyIdentityCanBeUpdated() : void
+    {
+        $this->createInitialRankingWithRelatedEntities();
+        $this->modifyRanking();
+        $this->verifyRanking();
+    }
+
+    private function createInitialRankingWithRelatedEntities() : void
+    {
+        $team    = new GH7062Team(self::TEAM_ID);
+        $season  = new GH7062Season(self::SEASON_ID);
+        $ranking = $season->start([$team]);
+
+        $this->_em->persist($team);
+        $this->_em->persist($season);
+        $this->_em->persist($ranking);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        foreach ($ranking->getPositions() as $position) {
+            self::assertFalse($position->hasPoints());
+        }
+    }
+
+    private function modifyRanking() : void
+    {
+        /** @var GH7062Ranking $ranking */
+        $ranking = $this->_em->find(GH7062Ranking::class, self::SEASON_ID);
+
+        $ranking->everybodyWins();
+
+        foreach ($ranking->getPositions() as $position) {
+            self::assertTrue($position->hasPoints());
+        }
+
+        $this->_em->flush();
+        $this->_em->clear();
+    }
+
+    private function verifyRanking() : void
+    {
+        /** @var GH7062Season $season */
+        $season = $this->_em->find(GH7062Season::class, self::SEASON_ID);
+        self::assertInstanceOf(GH7062Season::class, $season);
+
+        $ranking = $season->getRanking();
+        self::assertInstanceOf(GH7062Ranking::class, $ranking);
+
+        foreach ($ranking->getPositions() as $position) {
+            self::assertTrue($position->hasPoints());
+        }
+    }
+}
+
+/**
+ * Simple Entity whose identity is defined through another Entity (Season)
+ *
+ * @Entity
+ * @Table(name="soccer_rankings")
+ */
+class GH7062Ranking
+{
+    /**
+     * @Id
+     * @OneToOne(targetEntity=GH7062Season::class, inversedBy="ranking")
+     * @JoinColumn(name="season", referencedColumnName="id")
```

Comment:
@Ocramius @guilhermeblanco this test revealed a nice unrelated issue (when loading things with L2C the identifier flattener tries to use the [column name in an array that doesn't have that index](https://travis-ci.org/doctrine/doctrine2/jobs/346051424#L959)).
Forcing the entity to use the name of the field as the name of the column was the only way to isolate the issue that @mariusklocke described... what are your thoughts about this workaround?

Reply:
@lcobucci as long as we can open a new issue with the new problem, I'm good to isolate issues in 2 separate PRs. =)

Reply:
@guilhermeblanco that's the idea

Reply:
Yeah, that's a separate issue - let's not creep it into here.

Reply:
https://github.com/doctrine/doctrine2/issues/7096 created to handle that